.github/workflows/java-client-release-to-maven.yml

```diff
@@ -0,0 +1,42 @@
+name: Release Java Client to Maven
```

Comment:
Right, this is the release job. Do you have a test workflow? Or will this be separate as it did not exist before (and this is an integration effort)?

Reply:
I looked into other existing clients and found that tests for external clients are not run as part of the CI. I can add a step to run unit tests in the release pipeline if you suggest.

Reply:
@EnricoMi added a test workflow.